src/Missings.jl

```diff
@@ -441,29 +471,7 @@ julia> coalesce.([missing, 1, missing], [0, 10, 5])
 
 ```
 """
-coalesce(x) = x
-coalesce(x, y...) = ifelse(x !== missing, x, coalesce(y...))
-
-"""
-    levels(x)
-
-Return a vector of unique values which occur or could occur in collection `x`,
-omitting `missing` even if present. Values are returned in the preferred order
-for the collection, with the result of [`sort`](@ref) as a default.
-
-Contrary to [`unique`](@ref), this function may return values which do not
-actually occur in the data, and does not preserve their order of appearance in `x`.
-"""
-function levels(x)
-    T = Missings.T(eltype(x))
-    levs = convert(AbstractArray{T}, filter!(!ismissing, unique(x)))
-    if method_exists(isless, Tuple{T, T})
-        try; sort!(levs); end
-    end
-    levs
-end
-
-# Deprecations
-@deprecate skip(itr) skipmissing(itr) false
+Compat.coalesce(x::Missing) = missing
```

Comment:
This should only be defined before Base includes `coalesce`, or it's going to overwrite the equivalent Base method.

Reply:
That assumes that Compat already defines `coalesce(x::Missing)`, which it currently doesn't (it's commented out). It really boils down to whether the actual compat definition is in Missings or Compat.

Reply:
Right, but I was speaking about a different issue, which is that recent Julia 0.7 versions define that method, and that `Compat.coalesce = Base.coalesce` there.

Reply:
I don't think so? I don't see any overwrite happening on latest master?

Reply:
Easy enough to just wrap this in `is !isdefined(Base, :coalesce)` regardless.